lib/rspec/mocks/error_generator.rb

```diff
@@ -291,6 +303,19 @@ def format_received_args(*args)
         args.empty? ? "(no args)" : "(" + received_arg_list(*args) + ")"
       end
 
+      def format_unexpected_with_args(*args)
+        with_args = Hash[Array(args.first).group_by { |x| x }.map { |k, v| [k, v.count] }]
+        received_with_arg_list(*with_args)
+      end
+
+      def received_with_arg_list(*args)
+        list = args.map do |a, i|
+          times_count = args.size > 1 ? '(' + times(i) + ')' : ''
+          ["(#{received_arg_list(*a)})", "#{times_count}"].join(" ")
```

Comment:
Maybe we can have it only display the "n times" bit only if the message was received multiple times -- so any time that `(1 time)` would be displayed, we can suppress it (since that's what a user would normally assume and it's not needed in that case).  Then there's no need for an extra option.